File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/GroupStaleDevices.java
package org.whispersystems.signalservice.internal.push;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Represents the body of a 410 response from the service during a sender key send.
 */
public class GroupStaleDevices {

  @JsonProperty
  private String uuid;

  @JsonProperty
  private StaleDevices devices;

  public String getUuid() {
    return uuid;
  }

  public StaleDevices getDevices() {
    return devices;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/GroupStaleDevices.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.internal.push

/**
 * Represents the body of a 410 response from the service during a sender key send.
 */
data class GroupStaleDevices(
  val uuid: String,
  val devices: StaleDevices
)
